ansible/roles/hypothesis/templates/nginx/hypothesis.conf

```diff
@@ -1,3 +1,48 @@
+{% if use_https %}
+server {
+  listen 80;
+  server_name hypothesis.{{ liquid_domain }};
+
+  location /.well-known/acme-challenge {
+    alias /var/lib/liquid/https/acme-challenge;
+  }
+
+  location / {
+    return 301 https://$host$request_uri;
+  }
+}
+
+
+server {
+  listen 443 ssl;
+  server_name hypothesis.{{ liquid_domain }};
+
+  ssl_certificate /var/lib/liquid/https/certs/hypothesis.{{ liquid_domain }}/fullchain.pem;
+  ssl_certificate_key /var/lib/liquid/https/certs/hypothesis.{{ liquid_domain }}/privkey.pem;
+  include /var/lib/liquid/https/nginx/https.conf;
+
+  location / {
+    proxy_pass http://localhost:46124;
+    proxy_set_header Host $host;
+    proxy_set_header X-Forwarded-Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+  }
+
+  location /ws {
+    proxy_pass http://localhost:46125;
```

Comment:
So we're duplicating a bunch of text here. You can use [jinja macros](http://jinja.pocoo.org/docs/2.9/templates/#macros) for this.

Reply:
I'd prefer to keep these configs as simple as possible. Especially when there's something that needs debugging in there.

Reply:
Yes but they're complex enough to diverge accidentally. Let's see how they look like with macros and maybe you like them. Should I make another branch to experiment, or do you want to?

Reply:
Sure, let's experment on another branch. I'll triage this branch first.